script/tool/lib/src/publish_plugin_command.dart

```diff
@@ -111,7 +135,74 @@ class PublishPluginCommand extends PluginCommand {
     }
     _print('Local repo is ready!');
 
-    final Directory packageDir = _getPackageDir(package);
+    if (all) {
+      await _publishAllPackages(
+          remote: remote,
+          remoteUrl: remoteUrl,
+          shouldPushTag: shouldPushTag,
+          baseGitDir: baseGitDir);
+    } else {
+      await _publishAndReleasePackage(
+          packageDir: _getPackageDir(package),
+          remote: remote,
+          remoteUrl: remoteUrl,
+          shouldPushTag: shouldPushTag);
+    }
+    await _finishSuccesfully();
+  }
+
+  Future<void> _publishAllPackages(
+      {String remote,
+      String remoteUrl,
+      bool shouldPushTag,
+      GitDir baseGitDir}) async {
+    final List<String> packagesReleased = <String>[];
+    final GitVersionFinder gitVersionFinder = await retrieveVersionFinder();
+    final List<String> changedPubspecs =
+        await gitVersionFinder.getChangedPubSpecs();
+    if (changedPubspecs.isEmpty) {
+      _print('No version updates in this commit, exiting...');
+      return;
+    }
+    _print('Getting existing tags...');
+    final io.ProcessResult existingTagsResult =
+        await baseGitDir.runCommand(<String>['tag', '--sort=-committerdate']);
+    final List<String> existingTags = (existingTagsResult.stdout as String)
+        .split('\n')
+          ..removeWhere((element) => element == '');
+    for (final String pubspecPath in changedPubspecs) {
+      final File pubspecFile =
+          fileSystem.directory(baseGitDir.path).childFile(pubspecPath);
+      if (!pubspecFile.existsSync()) {
+        printErrorAndExit(
+            errorMessage:
+                'Fatal: The pubspec file at ${pubspecFile.path} does not exist.');
+      }
+      final bool needsRelease = await _checkNeedsRelease(
+          pubspecPath: pubspecPath,
+          pubspecFile: pubspecFile,
+          gitVersionFinder: gitVersionFinder,
+          existingTags: existingTags);
+      if (!needsRelease) {
+        continue;
+      }
+      _print('\n');
+      await _publishAndReleasePackage(
+          packageDir: pubspecFile.parent,
+          remote: remote,
+          remoteUrl: remoteUrl,
+          shouldPushTag: shouldPushTag);
+      packagesReleased.add(pubspecFile.parent.basename);
+      _print('\n');
+    }
+    _print('Packages released: ${packagesReleased.join(', ')}');
+  }
+
+  Future<void> _publishAndReleasePackage(
```

Comment:
`_publishAndTagPackage`? I would consider publishing to be releasing.

Wouldn't hurt to add a comment to this method saying what it does.